syntax="proto3";

message IndexWord {
    bytes word = 1;
};

message IndexImageRequest {
    string image_id = 1;
}

message IndexImageResponse {

};

message GetBagOfWordsRequest {
    string image_id = 1;
};

message GetBagOfWordsResponse {
    repeated IndexWord words = 1;
}

message ClosestNRequest{
    string image_id = 1;
    uint32 N = 2;
};

message ClosestNResponse {
    repeated string image_ids = 1;
}

service VisualIndexingService {
   rpc IndexImage(IndexImageRequest) returns (IndexImageResponse);
   rpc GetBagOfWords(GetBagOfWordsRequest) returns (GetBagOfWordsResponse);
   rpc ClosestN(ClosestNRequest) returns (ClosestNResponse);
};